x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/random/RandomRankBuilder.java

```
@@ -0,0 +1,164 @@
+        PARSER.declareFloat(optionalConstructorArg(), MIN_SCORE_FIELD);
+    }
+
+    private final String field;
+    private final Float minScore;
+
+    public RandomRankBuilder(int rankWindowSize, String field, Float minScore) {
+        super(rankWindowSize);
+        this.field = field;
+        this.minScore = minScore;
+    }
+
+    public RandomRankBuilder(StreamInput in) throws IOException {
+        super(in);
+        // rankWindowSize deserialization is handled by the parent class RankBuilder
+        this.field = in.readOptionalString();
+        this.minScore = in.readOptionalFloat();
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.RANDOM_RERANKER_RETRIEVER;
+    }
+
+    @Override
+    public void doWriteTo(StreamOutput out) throws IOException {
+        // rankWindowSize serialization is handled by the parent class RankBuilder
+        out.writeOptionalString(field);
+        out.writeOptionalFloat(minScore);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        // rankWindowSize serialization is handled by the parent class RankBuilder
+        builder.field(FIELD_FIELD.getPreferredName(), field);
+        if (minScore != null) {
+            builder.field(MIN_SCORE_FIELD.getPreferredName(), minScore);
+        }
+    }
+
+    @Override
+    public boolean isCompoundBuilder() {
+        return false;
+    }
+
+    @Override
+    public Explanation explainHit(Explanation baseExplanation, RankDoc scoreDoc, List<String> queryNames) {
+        if (scoreDoc == null) {
+            return baseExplanation;
+        }
+        if (false == baseExplanation.isMatch()) {
+            return baseExplanation;
+        }
+
+        assert scoreDoc instanceof RankFeatureDoc : "ScoreDoc is not an instance of RankFeatureDoc";
```

Do we ever expect this assertion to fail in production? If so, should we throw an exception instead?

---

I would not expect this to fail in production and this also uses the convention set by the text similarity rerank retriever.